openssl/src/cms.rs

```diff
@@ -219,14 +219,66 @@ impl CmsContentInfo {
             Ok(CmsContentInfo::from_ptr(cms))
         }
     }
+
+    /// Verify this CmsContentInfo's signature, given a stack of certificates
+    /// in certs, an X509 store in store. If the signature is detached, the
+    /// data can be passed in data. The data sans signature will be copied
+    /// into output_data if it is present.
+    ///
+    /// OpenSSL documentation at [`CMS_verify`]
+    ///
+    /// [`CMS_verify`]: https://www.openssl.org/docs/manmaster/man3/CMS_verify.html
+    #[allow(clippy::redundant_clone)]
```

Comment:
What is being redundantly cloned?

Reply:
*out_data = out_bio.get_buf().to_vec().clone();

I believe that the clone is not redundant, because the source object is not released by rust code but by OpenSSL (see the next comment].

Reply:
Vec owns its memory. There is no difference between the one created by to_vec and the one created by clone.

Reply:
OK. I got confused by slice::into_vec(] documentation.